src/app/modules/main/wallet_section/accounts/model.nim

```diff
@@ -173,34 +191,71 @@ QtObject:
     of ModelRole.CanSend:
       result = newQVariant(item.canSend())
 
+  proc updateBalance*(self: Model, address: string, balance: CurrencyAmount, assetsLoading: bool) =
+    let i = self.findAccountIndex(address)
+    if i < 0:
+      return
+    self.items[i].updateBalance(balance, assetsLoading)
```

Comment:
WDYT of adding 2 `set` methods on the `item` object instead of the `updateBalance` to close the API for modification in this regard? In the case of `updateBalance ` the `model` assumes that only these 2 roles changed and will emit the `dataChanged` signal. But that's just an assumption from the `model` implementation based on the current state.